### PR TITLE
650-display of the gdt transactions

### LINKED ISSUE #650: [FEATURE] View for the GDT Transaction List
## GDT Transaction List View 
Gradido has a system named Gradido Transform that is for people who actively helps Gradido they get a GDT what is a promise of amount of feature GDD that does not decay.

- [x] Definition where this should be shown (Maybe on a new page?)
- [ ] Definition of how it should look like
- [ ] Get datas from Apollo Server (Blocker?)
- [ ] Implementation of the new layout

- fixes #657

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
The transaction list of Gradido and Gradido Transform is displayed in a tab system. 
The style is approximately the same in both lists. 
Two types are currently intercepted
1. conversion EURO / GDT
2. GDT + Event 21 only 

### Issues
- fixes #650

 
FireShot Capture 189 - Gradido Account - localhost

### CODE REVIEW

**Comment by Mogge** [comment]
Ist es nicht besser einen neuen Menü-Punkt für die GDT Transaktionen zu machen, statt diese ans Ende der GDD Transaktionen zu kleben? Wenn ja, dann schlage ich vor, die Daten nicht im Dashboard-Layout zu laden, sondern statt dessen direkt in der Komponente, die die Daten anzeigt.

**Comment by Mogge** [comment]
Ich schlage vor den API call für die GDT Transaktionen in der GDTTransactionList Komponente zu machen. Wir brauchen die Daten nicht auf top-level

**Comment by ogerly** [comment]
@Mogge  Unfortunately, I can't bake it

File: community_server/src/Template/AppRequests/list_transactions.ctp [hunk 1]
```diff
@@ -8,7 +8,6 @@
 
 $body['balance'] = $this->element('centToFloat', ['cent' => $body['balance'], 'precision' => 4]);
 $body['decay'] = $this->element('centToFloat', ['cent' => $body['decay'], 'precision' => 4]);
-$body['gdtSum'] = $this->element('centToFloat', ['cent' => $body['gdtSum'], 'precision' => 2]);
```
Comment (Mogge) [inline community_server/src/Template/AppRequests/list_transactions.ctp:11]: Please add this line of code again.

File: community_server/tests/TestCase/Controller/AppRequestControllerTest.php [hunk 2]
```diff
@@ -260,7 +260,6 @@ public function testListTransactions()
 	],
 	"transactionExecutingCount": 0,
 	"count": 7,
-	"gdtSum": 180000,
```
Comment (Mogge) [inline community_server/tests/TestCase/Controller/AppRequestControllerTest.php:263]: And this one.

File: frontend/public/json-example/admin_transactionGdt_list.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{"state":"success", "gdt":
```
Comment (Mogge) [inline frontend/public/json-example/admin_transactionGdt_list.json:1]: Is this file still needed? If not, please remove.

File: frontend/src/locales/de.json [hunk 4]
```diff
@@ -165,5 +167,13 @@
     "title": "Passwort zurücksetzen",
     "text": "Jetzt kannst du ein neues Passwort speichern, mit dem du dich zukünftig in der Gradido-App anmelden kannst.",
     "not-authenticated": "Leider konnten wir dich nicht authentifizieren. Bitte wende dich an den Support."
+  },
+  "gdt": {
+     "gdt-received":"Gradido Transform (GDT) erhalten",
+     "factor":"Factor",
```
Comment (Mogge) [inline frontend/src/locales/de.json:173]: Faktor

File: frontend/src/views/Pages/AccountOverview/GddTransactionList.vue [hunk 5]
```diff
@@ -24,8 +24,7 @@
                 {{ $t('form.memo') }}
               </small>
             </div>
-            <div v-if="decay">
-              <br />
+            <div v-if="decay" class="text-sm">
```
Comment (Mogge) [inline frontend/src/views/Pages/AccountOverview/GddTransactionList.vue:27]: Is this change wanted or by accident?

File: frontend/src/views/Pages/AccountOverview/GdtTransactionList.vue [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        }
+      this.throwError('no icon to given type')
+    },
+    throwError(msg) {
+      throw new Error(msg)
+    },
+    async updateGdt() {
+      const result = await communityAPI.transactionsgdt(this.$store.state.sessionId)
+      this.transactionsGdt = result.result.data.gdtEntries
+      this.transactionGdtCount = result.result.data.count
```
Comment (Mogge) [inline frontend/src/views/Pages/AccountOverview/GdtTransactionList.vue:162]: Check for success first? Error handling?

File: frontend/src/views/Pages/UserProfileTransactionList.spec.js [hunk 7]
```diff
@@ -3,11 +3,31 @@ import UserProfileTransactionList from './UserProfileTransactionList'
 
 const localVue = global.localVue
 
+const mutationObserverMock = jest.fn(function MutationObserver(callback) {
```
Comment (Mogge) [inline frontend/src/views/Pages/UserProfileTransactionList.spec.js:6]: I added this one hoping to get lose of the warnings. But it did not work out :(

File: frontend/src/views/Pages/UserProfileTransactionList.spec.js [hunk 8]
```diff
@@ -37,5 +57,9 @@ describe('UserProfileTransactionList', () => {
         expect.arrayContaining([expect.arrayContaining([{ firstPage: 2, items: 25 }])]),
       )
     })
+
+    it('renders the transaction gradido transform table', () => {
```
Comment (Mogge) [inline frontend/src/views/Pages/UserProfileTransactionList.spec.js:61]: Great! This is enough!

**Review by Mogge — CHANGES_REQUESTED** [review]
We are very close! Good work

File: frontend/public/json-example/admin_transactionGdt_list.json (hunk 3 above)
Comment (ogerly) [reply frontend/public/json-example/admin_transactionGdt_list.json:1]: removed

File: frontend/src/locales/de.json (hunk 4 above)
Comment (ogerly) [reply frontend/src/locales/de.json:173]: geändert

File: frontend/src/views/Pages/AccountOverview/GddTransactionList.vue (hunk 5 above)
Comment (ogerly) [reply frontend/src/views/Pages/AccountOverview/GddTransactionList.vue:27]: as can be seen in the picture ... when the transaction list is queried ... the decay information comes first. 
this has the TYPE 'decay'.  
when this first element is created, there should be no empty line. 
maybe there is a nicer way to solve this. but i have it clean for now. 
FireShot Capture 188 - Gradido Account - localhost

**Review by Mogge — CHANGES_REQUESTED** [review]
Still missing gdtSum in TransactionList

**Comment by ogerly** [comment]
sieht gute aus .. GDT wird angezeigt

**Review by Mogge — APPROVED** [review]
